server/src/main/java/org/elasticsearch/index/mapper/RuntimeField.java

```diff
@@ -200,18 +216,7 @@ RuntimeField parse(String name, Map<String, Object> node,
      */
     static Map<String, MappedFieldType> collectFieldTypes(Collection<RuntimeField> runtimeFields) {
         return runtimeFields.stream()
-            .flatMap(runtimeField -> {
-                List<String> names = runtimeField.asMappedFieldTypes().stream().map(MappedFieldType::name)
-                    .filter(name -> name.equals(runtimeField.name()) == false
-                        && (name.startsWith(runtimeField.name() + ".") == false
-                        || name.length() > runtimeField.name().length() + 1 == false))
-                    .collect(Collectors.toList());
-                if (names.isEmpty() == false) {
-                    throw new IllegalStateException("Found sub-fields with name not belonging to the parent field they are part of "
-                        + names);
-                }
-                return runtimeField.asMappedFieldTypes().stream();
-            })
+            .flatMap(runtimeField -> runtimeField.asMappedFieldTypes().stream())
```

Comment:
I was not sure why this had changed. I left it as -is for now. Do you recall why you removed the check?